smartrunner_driver/src/rosnode/smartrunner_node.cpp

```diff
@@ -269,15 +467,45 @@ void sr3dnode::getScanDataPointCloud2(const ros::TimerEvent &e)
             }
         }
         status = vsx_ReleaseDataContainer(&dch);
-        }
-    sensor_msgs::convertPointCloudToPointCloud2(scanmsg,scanmsg2);
-    
-    scan_publisher_.publish(scanmsg2);   
+    } 
+    sensor_msgs::convertPointCloudToPointCloud2(scanmsg,scanmsg2);  
```

Comment:
The PointCloud2-message organizes the points in a 2-dimensional array, just as the 3D images from the SR3D are organized. This information is lost if convertPointCloudToPointCloud2 is used.

Can we create a PointCloud2-object directly?